Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityInteractionMapping.cs

```diff
@@ -372,16 +372,9 @@ public float FloatData
                     Debug.LogError($"SetFloatValue is only valid for AxisType.SingleAxis InteractionMappings\nPlease check the {inputType} mapping for the current controller");
                 }
 
-                if (invertXAxis)
-                {
-                    Changed = !floatData.Equals(value * -1f);
-                    floatData = value * -1f;
-                }
-                else
-                {
-                    Changed = !floatData.Equals(value);
-                    floatData = value;
-                }
+                float newValue = (invertXAxis ? -1f : 1f) * value;
+                Changed = newValue != 0f || newValue != floatData;
```

Comment:
I'm not very familiar with floating point comparison issues in C# - I know that with C++ this sort of comparison is dangerous because you could get floating point comparison issues (i.e. something that is very near zero reports as zero, something)

I'm curious if you know about this stuffs

Reply:
I don't have additional information there but I'd be very surprised if floating point operations in C# were implemented differently to C/C++ in this regard. If `value` is exactly zero, `newValue` should be too and compare as equal against `0f`. Does that address your concern?